pkg/chartutil/requirements.go

```diff
@@ -106,3 +116,144 @@ func LoadRequirementsLock(c *chart.Chart) (*RequirementsLock, error) {
 	r := &RequirementsLock{}
 	return r, yaml.Unmarshal(data, r)
 }
+
+// ProcessRequirementsConditions disables charts based on condition path value in values
+func ProcessRequirementsConditions(reqs *Requirements, cvals Values) {
+	var cond string
+	var conds []string
+
+	if reqs != nil && len(reqs.Dependencies) > 0 {
+		for _, r := range reqs.Dependencies {
+			var hasTrue, hasFalse bool
+			cond = string(r.Condition)
+			// check for list
+			if len(cond) > 0 {
+				if strings.Contains(cond, ",") {
+					conds = strings.Split(strings.TrimSpace(cond), ",")
+				} else {
+					conds = []string{strings.TrimSpace(cond)}
+				}
+				for _, c := range conds {
+					if len(c) > 0 {
+						// retrieve value
+						vv, err := cvals.PathValue(c)
+						if err == nil {
+							if vv.(bool) {
+								hasTrue = true
+							}
+							if !vv.(bool) {
+								hasFalse = true
+							}
+						} else {
+							if _, ok := err.(ErrNoValue); !ok {
+								// this is a real error
+								log.Printf("Warning: PathValue returned error %v", err)
+							}
+						}
+						if vv != nil {
+							// got first value, break loop
+							break
+						}
+					}
+				}
+				if !hasTrue && hasFalse {
+					r.Enabled = false
+				} else {
+					if hasTrue {
+						r.Enabled = true
+					}
+				}
+			}
+
+		}
+	}
+}
+
+// ProcessRequirementsTags disables charts based on tags in values
+func ProcessRequirementsTags(reqs *Requirements, cvals Values) {
+	vt, err := cvals.Table("tags")
+	if err != nil {
+		return
+
+	}
+	if reqs != nil && len(reqs.Dependencies) > 0 {
+		for _, r := range reqs.Dependencies {
+			if len(r.Tags) > 0 {
+				tags := r.Tags
+
+				var hasTrue, hasFalse bool
+				for _, k := range tags {
+					if b, ok := vt[k]; ok {
+						if b.(bool) {
+							hasTrue = true
+						}
+						if !b.(bool) {
+							hasFalse = true
+						}
+					}
+				}
+				if !hasTrue && hasFalse {
+					r.Enabled = false
+				} else {
+					if hasTrue || !hasTrue && !hasFalse {
+						r.Enabled = true
+					}
+				}
+
+			}
+		}
+	}
+}
+
+// ProcessRequirementsEnabled removes disabled charts from dependencies
+func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
+	reqs, err := LoadRequirements(c)
+	if err != nil {
+		// if not just missing requirements file, return error
+		if nerr, ok := err.(ErrNoRequirementsFile); !ok {
+			return nerr
+		} else {
+			// no requirements to process
+			return nil
+		}
+	}
+	// set all to true
+	for _, lr := range reqs.Dependencies {
+		lr.Enabled = true
+	}
+	cvals, err := CoalesceValues(c, v)
+	if err != nil {
+		return err
+	}
+	// flag dependencies as enabled/disabled
+	ProcessRequirementsTags(reqs, cvals)
+	ProcessRequirementsConditions(reqs, cvals)
+
+	// make a map of charts to remove
+	rm := map[string]bool{}
+	for _, r := range reqs.Dependencies {
+		if !r.Enabled {
+			// remove disabled chart
+			rm[r.Name] = true
+		}
+	}
+	// don't keep disabled charts in new slice
+	cd := c.Dependencies[:0]
```

Comment:
If I remember right, you'll be storing a reference to the old underlying array here with the current way you are doing it, why not just do (could be wrong on some types here) `cd := []*Dependency` or the very slightly more efficient `cd := make([]*Dependency, len(c.Dependencies) - len(rm))`?